web/src/routes/news/containers/LocalNewsPage.js

```diff
@@ -25,52 +27,71 @@ type PropsType = {|
   path: string
 |}
 
-export class LocalNewsPage extends React.Component<PropsType> {
-  renderLocalNewsElement = (city: string, language: string) => (localNewsItem: LocalNewsModel) => {
-    const { id, title, message, timestamp } = localNewsItem
+export const LocalNewsPage = ({
+  localNews,
+  city,
+  cities,
+  areCitiesFetching,
+  path,
+  language,
+  t
+}: PropsType) => {
+  const formatter = useContext(DateFormatterContext)
+  const renderLocalNewsElement = (city: string, language: string) => (localNewsItem: LocalNewsModel) => {
+    const {
+      id,
+      title,
+      message,
+      timestamp
+    } = localNewsItem
     return <NewsElement
       title={title}
       content={message}
       timestamp={timestamp}
       key={id}
-      link={new LocalNewsDetailsRouteConfig().getRoutePath({ city, language, id })}
-      t={this.props.t}
-      language={language}
+      link={new LocalNewsDetailsRouteConfig().getRoutePath({
+        city,
+        language,
+        id
+      })}
+      t={t}
+      formatter={formatter}
       type={LOCAL_NEWS}
     />
   }
 
-  render () {
-    const { localNews, city, cities, areCitiesFetching, path, language, t } = this.props
-
-    if (areCitiesFetching) {
-      return <LoadingSpinner />
-    }
-
-    const currentCity: ?CityModel = cities && cities.find(cityElement => cityElement.code === city)
-    if (!currentCity) {
-      return <FailureSwitcher error={new CityNotFoundError()} />
-    } else if (!currentCity.pushNotificationsEnabled) {
-      const error = new NotFoundError({ type: 'category', id: path, city: city, language })
-      return <FailureSwitcher error={error} />
-    }
+  if (areCitiesFetching) {
+    return <LoadingSpinner />
+  }
 
-    return (
-      <NewsTabs type={LOCAL_NEWS}
-                city={city}
-                tunewsEnabled={currentCity.tunewsEnabled}
-                localNewsEnabled={currentCity.pushNotificationsEnabled}
-                t={t}
-                language={language}>
-        <LocalNewsList
-          items={localNews}
-          noItemsMessage={t('currentlyNoNews')}
-          renderItem={this.renderLocalNewsElement(city, language)}
-          city={city}
-        />
-      </NewsTabs>
-    )
+  const currentCity: ?CityModel = cities && cities.find(cityElement => cityElement.code === city)
```

Comment:
cities should always be defined here shouldn't it?

Reply:
find can return undefined